base/hexadecimal.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+namespace principia {
+namespace base {
+
+// The result is upper-case.  Either |&input[input_size] <= output| or
+// |&output[input_size] <= input| must hold.  |output_size| must be at least
+// twice |input_size|.
+void HexadecimalEncode(uint8_t const* input, size_t input_size,
```

Comment:
I'd use `int64_t` instead of `size_t` for the sizes.  Just say no to unsigned types.